### PR TITLE
Marriage Fix - Lynda

### PR DESCRIPTION
**Pull request by EeFo17**
Marriage is not working.

# Description

Now you need a wedding ring and a wedding dress, Lynda add wedding dresses and rings with a special description.

### **Actual**

At the moment, the only problem is removing the outfit in divorce. Only one partner loses outfit.

### **Expected**

After divorce players should loses wedding outfits.

## Fixes

\# (issue)

## Type of change

Please delete options that are not relevant.

  - [ x ] Bug fix (non-breaking change which fixes an issue)
  - [ x ] New feature (non-breaking change which adds functionality)

**Test Configuration**:

  - Server Version: 1.0.5
  - Client: 12.72
  - Operating System: Ubuntu 

## Checklist

  - [ ] My code follows the style guidelines of this project
  - [ ] I have performed a self-review of my own code
  - [ ] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [ ] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by EeFo17** [comment]
If anyone has any ideas on how to improve, please leave a comment.

**Comment by doenteverde** [comment]
only one player is winning the outfit

File: data/events/scripts/player.lua [hunk 1]
```diff
@@ -199,6 +199,11 @@ function Player:onLook(thing, position, distance)
 			end
 		end
 	end
+		if LOOK_MARRIAGE_DESCR and thing:isCreature() then
+        if thing:isPlayer() then
+            description = description .. self:getMarriageDescription(thing)
```
Comment (worthdavi) [inline data/events/scripts/player.lua:204]: self:getMarriageDescription() is calling getPlayerSpouse(), getPlayerMarriageStatus() and getPlayerNameById()
3 db queries in a onLook function

I think this function would be way easier to make in player.cpp/h

Comment (doenteverde) [reply data/events/scripts/player.lua:204]: How cool are you going to post or just said?  
kk

Comment (worthdavi) [reply data/events/scripts/player.lua:204]: Sorry, I don't get you. But if you meant if am I going to post some code, no. He asked for a hint/advice to improve the code, and I gave mine.

File: data/npc/scripts/lynda.lua [hunk 2]
```diff
@@ -126,44 +126,96 @@ local function confirmWedding(cid, message, keywords, parameters, node)
...
-        npcHandler:say('Your partner didn\'t accept your proposal, yet', cid)
+        msg = msg .. ' What brings you here? Wanting to {marry} someone?'
+    end
+    npcHandler:say(msg,cid)
+    npcHandler:addFocus(cid)
+    return false
+end
+
+local function tryEngage(cid, message, keywords, parameters, node)
+    if(not npcHandler:isFocused(cid)) then
```
Comment (dudantas) [inline data/npc/scripts/lynda.lua:143]: These functions no longer exist, the lib has been modified

File: data/npc/scripts/lynda.lua [hunk 3]
```diff
@@ -126,44 +126,96 @@ local function confirmWedding(cid, message, keywords, parameters, node)
...
+    if (playerStatus == MARRIED_STATUS) then
+        msg = msg .. ' I see that you are a happily married ' .. ((player:getSex() == PLAYERSEX_FEMALE) and 'woman' or 'man') .. '. What brings you here? Looking for a {divorce}?'
+    elseif (playerStatus == PROPOSED_STATUS) then
+        msg = msg .. ' You are still waiting for the wedding proposal you made to {' .. (getPlayerNameById(playerSpouse)) .. '}. Would you like to {remove} it?'
     else
-        npcHandler:say('Your partner didn\'t accept your proposal, yet', cid)
+        msg = msg .. ' What brings you here? Wanting to {marry} someone?'
+    end
+    npcHandler:say(msg,cid)
+    npcHandler:addFocus(cid)
```
Comment (dudantas) [inline data/npc/scripts/lynda.lua:138]: Same

**Review by dudantas — CHANGES_REQUESTED** [review]
You need to edit the npc to use npcs Lua new lib in revscript

**Comment by majestyotbr** [comment]
Pull request closed due to inactivity. If you want to finish, you can reopen the pull request.

**Pull request closed without merging**